core/src/v2/schema.rs

```diff
@@ -184,6 +184,8 @@ impl_type_simple!(
 #[cfg(feature = "uid")]
 impl_type_simple!(uuid::Uuid, DataType::String, DataTypeFormat::Uuid);
 
+pub type SchemaSecurityDefs = &'static [(&'static str, &'static [&'static str])];
```

Comment:
I also don't like that we have this complex structure. Either way, all of this happens at runtime. So, we could just have an associated function to emit the information we need (just like `raw_schema` function).